### ISSUE TITLE
Dynamically importing CommonJS produces empty bundle

### ISSUE THREAD

**Issue by mischnic** [parcel-bundler/parcel#3586]
Dynamically importing a commonjs module from a `esmodule` asset produces an empty bundle.

```html
<script type="module" src="./index.js"></script>
```
```js
// index.js
import("./commonjs.js").then(v => console.log(v));
```
```js
// commonjs.js
module.exports = 2;
```

Output (why is that even called `...exports`?):
```js
var $c0ee4c22c4f80f14efb8476eb9807b50$exports = {};
$c0ee4c22c4f80f14efb8476eb9807b50$exports = import('' + './commonjs.e4b0f555.js');
$c0ee4c22c4f80f14efb8476eb9807b50$exports.then(v => console.log(v));
```

`./commonjs.e4b0f555.js` is empty, without `treeShake`, the output is
```js
// ...
var $dfc1ea661454a5b57fd7e754dd151b32$exports = {};
$dfc1ea661454a5b57fd7e754dd151b32$exports = 2;
```

We check `b.env.outputFormat === 'esmodule'`, but this value is just inherited from that parent asset.

The root cause is of course that we don't/can't translate commonjs to esmodules, but this is a case where this might be unexpected.

cc @devongovett

**Comment by mischnic** [comment]
I guess we should for now throw an error if the entry asset of the dynamic bundle has CommonJS exports?

**Comment by wangkewei1993** [comment]
hi

**Comment by devongovett** [comment]
`import()` returns the entire module namespace. Should we just map the common JS `module.exports` object to that namespace if we detect it is CommonJS?

**Comment by mischnic** [comment]
> Should we just map the common JS module.exports object to that namespace if we detect it is CommonJS?

```js
const $module$exports = {
	function : () => {},
	constant: 2
};

export $module$exports; // !!
```

Exporting a JS object as a namespace isn't possible, either `export {prop: val, ...}` (but we don't know the properties at built time, that is the whole point of this) or `export default $module$exports`.

**Comment by mischnic** [comment]
Fixed by #3800
